src/ebonite/ext/onnx/model.py

```diff
@@ -0,0 +1,107 @@
+import contextlib
+import os
+from abc import abstractmethod
+from typing import Dict, List, Type, Union
+
+import onnx
+from onnx import ModelProto
+from pyjackson.decorators import cached_property
+
+from ebonite.core.analyzer import TypeHookMixin
+from ebonite.core.analyzer.model import BindingModelHook
+from ebonite.core.objects import ModelIO, ModelWrapper, Requirements
+from ebonite.core.objects.artifacts import Blobs, InMemoryBlob
+from ebonite.core.objects.wrapper import FilesContextManager
+from ebonite.utils.abc_utils import is_abstract_method
+from ebonite.utils.importing import module_importable
+
+_DEFAULT_BACKEND = 'onnxruntime'
+
+
+def set_default_onnx_backend(backend: Union[Type['ONNXInferenceBackend'], str]):
+    global _DEFAULT_BACKEND
+    if not isinstance(backend, str):
+        backend = backend.name
+    if backend not in ONNXInferenceBackend.subtypes:
+        raise ValueError(f'unknown onnx backend {backend}')
+    _DEFAULT_BACKEND = backend
+
+
+class ONNXModelIO(ModelIO):
+    FILENAME = 'model.onnx'
+
+    @contextlib.contextmanager
+    def dump(self, model: ModelProto) -> FilesContextManager:
+        yield Blobs({self.FILENAME: InMemoryBlob(model.SerializeToString())})  # TODO change to LazyBlob
+
+    def load(self, path):
+        return onnx.load(os.path.join(path, self.FILENAME))
+
+
+class ONNXInferenceBackend:
+    subtypes: Dict[str, Type['ONNXInferenceBackend']] = {}
+    name: str
+    requirements: List[str]
+
+    def __init__(self, wrapper: 'ONNXModelWrapper'):
+        self.wrapper = wrapper
+
+    def __init_subclass__(cls, **kwargs):
+        if not is_abstract_method(cls.run) and not is_abstract_method(cls.is_available):
+            if 'name' not in cls.__dict__ or 'requirements' not in cls.__dict__:
```

Comment:
Strange check, considering that these are fields of parent class. Isn't it easier to check if class is instance of ONNXInferenceBackend?